kpi/deployment_backends/mock_backend.py

```diff
@@ -236,38 +245,46 @@ def get_submission_validation_status_url(self, submission_id: int) -> str:
         )
         return url
 
-    def get_submissions(self,
-                        user: 'auth.User',
-                        format_type: str = INSTANCE_FORMAT_TYPE_JSON,
-                        submission_ids: list = [],
-                        **kwargs) -> list:
+    def get_submissions(
+        self,
+        user: 'auth.User',
+        format_type: str = SUBMISSION_FORMAT_TYPE_JSON,
+        submission_ids: list = [],
+        **mongo_query_params
+    ) -> list:
         """
-        Retrieves submissions whose `user` is allowed to access
+        Retrieve submissions that `user` is allowed to access.
+
         The format `format_type` can be either:
-        - 'json' (See `kpi.constants.INSTANCE_FORMAT_TYPE_JSON)
-        - 'xml' (See `kpi.constants.INSTANCE_FORMAT_TYPE_XML)
+        - 'json' (See `kpi.constants.SUBMISSION_FORMAT_TYPE_JSON)
+        - 'xml' (See `kpi.constants.SUBMISSION_FORMAT_TYPE_XML)
+
+        Results can be filtered by submission ids. Moreover MongoDB filters can
+        be passed through `mongo_query_params` to narrow down the results.
 
-        Results can be filtered on instance ids and/or MongoDB filters can be
-        passed through `kwargs`
+        If `user` has no access to these submissions or no matches are found,
+        an empty list is returned.
+        If `format_type` is 'json', a list of dictionary is returned.
+        Otherwise, if `format_type` is 'xml', a list of string is returned.
         """
 
-        kwargs['submission_ids'] = submission_ids
+        mongo_query_params['submission_ids'] = submission_ids
         params = self.validate_submission_list_params(user,
                                                       format_type=format_type,
-                                                      **kwargs)
+                                                      **mongo_query_params)
 
-        submissions, total_count = MongoHelper.get_instances(
+        mongo_cursor, total_count = MongoHelper.get_instances(
             self.mongo_userform_id, **params)
 
         # Python-only attribute used by `kpi.views.v2.data.DataViewSet.list()`
         self.current_submissions_count = total_count
 
         submissions = [
             MongoHelper.to_readable_dict(submission)
-            for submission in submissions
+            for submission in mongo_cursor
```

Comment:
We're not doing that at this time because the tests often rely on the original list of submissions being available validating results against expectations. This wouldn't work with a generator.